### PR TITLE
fix(cli/ssh): Avoid connection hang when workspace is stopped

### PR DESCRIPTION
**Pull request by mafredri**
Two issues are addressed here:
1. We were not detecting disconnects due to waiting for Stdin to close
   (disconnect would only propagate after entering input and failing to
   write to the connection).
2. In other scenarios, where the connection drop is not detected, we now
   also watch workspace status and drop the connection when a workspace
   reaches the stopped state.

Fixes: https://github.com/coder/jetbrains-coder/issues/199

Refs: #6180, #6175

### CODE REVIEW

File: cli/ssh.go [hunk 1]
```diff
@@ -100,17 +100,62 @@ func (r *RootCmd) ssh() *clibase.Cmd {
...
+					case w, ok := <-wsWatch:
+						if !ok {
+							return
+						}
+
+						// Note, we only react to the stopped state here because we
+						// want to give the agent a chance to gracefully shut down
+						// during "stopping".
+						if w.LatestBuild.Status == codersdk.WorkspaceStatusStopped {
+							_, _ = fmt.Fprintf(inv.Stderr, "Workspace %q has stopped. Closing connection.\r\n", workspace.Name)
```
Comment (mafredri) [inline cli/ssh.go:135]: Yay or nay? This seems to be visible even non-verbose SSH is used (e.g. both `ssh` and `ssh -vvv`).

**Review by kylecarbs — APPROVED** [review]
Fantastic tests too!

**Review by code-asher — APPROVED** [review]
I tried it with Gateway and confirmed it fixes the workspace restart issue!  :tada: 

This is probably a dumb question but does this mean there could be some problem with a disconnect not getting detected or propagated?  Like if this is going over Tailscale for example I would imagine there is some kind of ping/pong that would normally close the connection after a timeout but in Gateway the proxy command hangs indefinitely which seems weird.  Maybe that is by design though to keep retrying in the background or something.

**Comment by code-asher** [comment]
Ahhhhhh that makes sense!  72 hours seems like an excessively long time but maybe that is to keep it alive over the weekend or something?  I tried adding `ServerAliveInterval` on the client side and that works, might add that to the plugin.

**Pull request merged**